src/v2/Components/NavBar/NavItem.tsx

```diff
@@ -67,81 +127,92 @@ export const NavItem: React.FC<NavItemProps> = ({
     }
   }
 
+  const containerRef = useRef<null | HTMLDivElement>(null)
+
+  // Close the subnav if it is open and our focus moves outside of it
+  useEffect(() => {
+    const checkFocus = () => {
+      if (!containerRef?.current) return // Not mounted
+      if (!isVisible) return // Isn't open
+      if (containerRef.current.contains(document.activeElement)) return // Focus is within sub-nav
+
+      setIsVisible(false)
+    }
+
+    document.addEventListener("focus", checkFocus, true)
+    return () => {
+      document.removeEventListener("focus", checkFocus)
+    }
+  }, [isVisible])
+
+  const handleClick = () => {
+    onClick && onClick()
+    !!Menu && setIsVisible(prevIsVisible => !prevIsVisible)
+    trackClick()
+  }
+
+  // Add a very short delay to avoid trapping the mouse and capturing intent
+  const openDelayRef = useRef<null | ReturnType<typeof setTimeout>>(null)
+  // Add a slightly longer delay to improve UX when moving at an angle off HitArea
+  const closeDelayRef = useRef<null | ReturnType<typeof setTimeout>>(null)
+
+  const handleMouseEnter = () => {
+    openDelayRef.current = setTimeout(() => {
+      setIsVisible(true)
+    }, 50)
+    closeDelayRef?.current && clearTimeout(closeDelayRef.current)
+  }
+
+  const handleMouseLeave = () => {
+    openDelayRef?.current && clearTimeout(openDelayRef.current)
+    closeDelayRef.current = setTimeout(() => {
+      setIsVisible(false)
+    }, 150)
+  }
+
   return (
-    <Box
-      position={showOverlay ? "relative" : "static"}
-      onMouseEnter={() => setIsVisible(true)}
-      onMouseLeave={() => setIsVisible(false)}
+    <Container
+      ref={containerRef as any}
+      // When fullscreen: position relative to outer container
+      // When not fullscreen: position relative to immediate container
+      position={menuAnchor === "full" ? undefined : "relative"}
+      onMouseEnter={handleMouseEnter}
+      onMouseLeave={handleMouseLeave}
+      {...rest}
     >
-      <Link
+      <HitArea
+        as={Menu ? Clickable : undefined}
```

Comment:
Here if it has a sub-menu we just use the `Clickable`, which is an actual `button` underneath. So we get focusable and keypress handling automatically.